### PR TITLE
[types] Define Address types and Authenticator

### PR DESCRIPTION
**Pull request by jolestar**
1. Define Address Types. Try to support MultiChainAddress. RoochAddress, EthereumAddress, BitcoinAddress.
2. Define Rooch transaction and Authenticator, and introduce the crypto library.

This is part of #72

### CODE REVIEW

**Comment by jolestar** [comment]
@wow-sven The cli maybe also need the crypto library.

**Comment by wow-sven** [comment]
> @wow-sven The cli maybe also need the crypto library.

Yes, I'm using it at the moment
[fastcrypto](https://github.com/MystenLabs/fastcrypto)

**Comment by jolestar** [comment]
> > @wow-sven The cli maybe also need the crypto library.
> 
> Yes, I'm using it at the moment [fastcrypto](https://github.com/MystenLabs/fastcrypto)

I just reuse the `starcoin-crypto`, you can replace it if the `fastcrypto` is more convenient.

**Review by wow-sven — APPROVED** [review]

**Review by baichuan3 — APPROVED** [review]

File: crates/rooch-types/src/address.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    fn try_from(value: MultiChainAddress) -> Result<Self, Self::Error> {
+        if value.coin.id != slip44::Coin::Ether.id() {
+            return Err(anyhow::anyhow!("coin id {} is invalid", value.coin.id));
+        }
+        Ok(Self(H160::from_slice(&value.address)))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct BitcoinAddress(pub bitcoin::Address);
```
[the remark this answers is not in the document]
Comment (jolestar) [reply crates/rooch-types/src/address.rs:187]: Try to support Address mapping first.

**Pull request merged**